lib/pacemaker/tests/pcmk_ticket/pcmk_ticket_delete_test.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/cib/internal.h>
+#include <crm/common/unittest_internal.h>
+#include <crm/common/xml.h>
+#include <pacemaker.h>
+
+static char *cib_path = NULL;
+
+static void
+cib_not_connected(void **state)
+{
+    xmlNode *xml = NULL;
+
+    /* Without any special setup, cib_new() in pcmk_ticket_delete will use the
+     * native CIB which means IPC calls.  But there's nothing listening for those
+     * calls, so signon() will return ENOTCONN.  Check that we handle that.
+     */
+    assert_int_equal(pcmk_ticket_delete(&xml, "ticketA", false), ENOTCONN);
+    pcmk__assert_validates(xml);
+    free_xml(xml);
+}
+
+static int
+setup_test(void **state)
+{
+    cib_path = pcmk__cib_test_copy_cib("tickets.xml");
+
+    if (cib_path == NULL) {
+        return -1;
+    }
+
+    return 0;
+}
+
+static int
+teardown_test(void **state)
+{
+    pcmk__cib_test_cleanup(cib_path);
+    cib_path = NULL;
+    return 0;
+}
+
+static void
+bad_arguments(void **state)
+{
+    xmlNode *xml = NULL;
+
+    assert_int_equal(pcmk_ticket_delete(NULL, "ticketA", false), EINVAL);
+
+    assert_int_equal(pcmk_ticket_delete(&xml, NULL, false), EINVAL);
+    pcmk__assert_validates(xml);
+    free_xml(xml);
+}
+
+static void
+unknown_ticket(void **state)
+{
+    xmlNode *xml = NULL;
+
+    assert_int_equal(pcmk_ticket_delete(&xml, "XYZ", false), ENXIO);
+    pcmk__assert_validates(xml);
+    free_xml(xml);
+    xml = NULL;
+
+    assert_int_equal(pcmk_ticket_delete(&xml, "XYZ", true), pcmk_rc_ok);
+    pcmk__assert_validates(xml);
+    free_xml(xml);
+}
+
+static void
+ticket_granted(void **state)
+{
+    xmlNode *xml = NULL;
+
+    assert_int_equal(pcmk_ticket_delete(&xml, "ticketB", false), EACCES);
```

Comment:
Shouldn't we test a force-deleted granted ticket too?

Reply:
Yeah, definitely.  I thought I'd already done that - not sure what happened to it.